src/_pytest/assertion/util.py

```diff
@@ -225,10 +227,21 @@ def _diff_text(left: str, right: str, verbose: int = 0) -> List[str]:
         left = repr(str(left))
         right = repr(str(right))
         explanation += ["Strings contain only whitespace, escaping them using repr()"]
-    explanation += [
-        line.strip("\n")
-        for line in ndiff(left.splitlines(keepends), right.splitlines(keepends))
-    ]
+
+    left_lines = left.splitlines(keepends)
+    right_lines = right.splitlines(keepends)
+
+    if any(
+        wcwidth(ch) <= -1
+        for ch in itertools.chain.from_iterable([x for x in left_lines + right_lines])
```

Comment:
Maybe `for ch in line for line in left_lines + right_lines` (or it's the other way around - I always forget) instead of itertools big guns?

Reply:
Changed to `for ch in [ch for lines in left_lines + right_lines for ch in lines]`.